src/transport/endian.rs

```diff
@@ -0,0 +1,116 @@
+/// The endianness to use for encoding header fields.
+///
+/// The encoding and serialization of message bodies is up to the application code,
+/// and it not affected by this configuration parameter.
+#[derive(Debug, Clone, Copy, Eq, PartialEq, Ord, PartialOrd, Hash)]
+pub enum Endian {
```

Comment:
How do you feel about adding `NativeEndian`?

Reply:
Bad: if both ends use `native-endian` on different platforms it could theoretically be different.

I'd rather be explicit what it is.

Reply:
On the other hand, it makes sense for peers using unix sockets.... Hmmm.

Reply:
We could have a constructor that instantiates to one of the two endiannesses..

Reply:
> We could even forgo `Endian` for unix transport and just use native endian by default.

That's not that easy though. You can use the `StreamTransport` with Unix stream sockets too.

Reply:
Ah yeah good point.

Reply:
Added NativeEndian anyway ^^